pkg/bpf/load.go

```diff
@@ -0,0 +1,133 @@
+//  Copyright 2022 Database Mesh Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package bpf
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/ringbuf"
+	"github.com/mlycore/log"
+)
+
+const (
+	SockFilter = "sock_filter.o"
+	TcPktMap   = "/sys/fs/bpf/tc/globals/my_pkt"
+)
+
+func load() error {
+	m, err := loadTcPktMap()
+	if err != nil {
+		return err
+	}
+
+	if err := loadSockFilter(m); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+type Objs struct {
+	Prog         *ebpf.Program `ebpf:"sql_filter"`
+	TailProg1    *ebpf.Program `ebpf:"sql_filter_1"`
+	FilterHelper *ebpf.Map     `ebpf:"filter_helper"`
+	Buf          *ebpf.Map     `ebpf:"buf"`
+	JmpTable     *ebpf.Map     `ebpf:"jmp_table"`
+	MyPktEvt     *ebpf.Map     `ebpf:"my_pkt_evt"`
+}
+
+func loadSockFilter(tcPkt *ebpf.Map) error {
+	spec, err := ebpf.LoadCollectionSpec(SockFilter)
+	if err != nil {
+		return err
+	}
+
+	var objs Objs
+	if err := spec.LoadAndAssign(&objs, nil); err != nil {
+		return err
+	}
+
+	if err = objs.JmpTable.Update(uint32(0), uint32(objs.TailProg1.FD()), ebpf.UpdateAny); err != nil {
+		return fmt.Errorf("jmptable err: %s", err)
+	}
+
+	reader, err := ringbuf.NewReader(objs.MyPktEvt)
+	if err != nil {
+		return err
+	}
+
+	for {
+		evt, query, err := readRecord(objs, reader)
+		if err != nil {
+			log.Infoln(err)
+			continue
+		}
+
+		evt.ClassId = calcQos(query)
+		if err := tcPkt.Update(uint32(0), &evt, ebpf.UpdateAny); err != nil {
+			log.Infoln(err)
```

Comment:
This would be log.Warnln or Errorln

Reply:
fix in [444f1b8](https://github.com/database-mesh/waterline/pull/7/commits/444f1b824fa3024aea1752c5f6f7c317d08afbdd)